lib/ui/pages/news_list/primary_news_list_item.dart

```diff
@@ -0,0 +1,116 @@
+import 'package:allthenews/ui/pages/news_list/primary_news_list_entity.dart';
+import 'package:flutter/material.dart';
+
+abstract class _Constants {
+  static const double newsImageRadius = 15;
+  static const double newsImageLayerOpacity = 0.7;
+  static const double authorImageRadius = 15;
+  static const double authorRightPadding = 8;
+  static const double stackItemPadding = 20;
+}
+
+class PrimaryNewsListItem extends StatelessWidget {
+  final PrimaryNewsListEntity news;
+
+  PrimaryNewsListItem({Key key, @required this.news})
+      : assert(news != null),
+        super(key: key);
+
+  @override
+  Widget build(BuildContext context) => Stack(
+        children: <Widget>[
+          ClipRRect(
+            borderRadius: BorderRadius.circular(_Constants.newsImageRadius),
+            child: Container(
+              color: Colors.black,
+              child: Opacity(
+                opacity: _Constants.newsImageLayerOpacity,
+                child: Image.network(
+                  news.imageUrl,
+                  //FIXME enter correct dimensions when filling a list with those items
+                  width: 300,
+                  height: 300,
+                  fit: BoxFit.fill,
+                ),
+              ),
+            ),
+          ),
+          Positioned(
+            top: 1,
+            right: 1,
+            child: _buildBookmarkIcon(),
+          ),
+          _buildNewsInfo(context),
+        ],
+      );
+
+  Positioned _buildNewsInfo(BuildContext context) => Positioned(
```

Comment:
Z tego co widziałem w googlowych samplach to konwencja dla metod tworzących jakiś widget jest taka, że nie określany jest specyficzny typ widgetu (tj. Padding, Align, itd.) tylko jako typ zwracany jest po prostu Wigdet i w sumie takiej konwencji bym się trzymał patrząc na to jak zaprojektowane jest budowanie UI - tj. każdy child/parent to Widget i ten poziom abstrakcji wystarcza dla frameworka ;)

Reply:
właśnie coś mi nie pasowało w tych sygnaturach :D